app/gui/view/debug_scene/cloud-dashboard/src/lib.rs

```diff
@@ -37,131 +37,483 @@
+            let grid_height = screen_size.y * GRID_HEIGHT_RATIO;
+            let grid_min_x = viewport_min_x;
+            let grid_max_y = viewport_min_y + grid_height;
+            let position = Vector2::new(grid_min_x, grid_max_y);
+            self.grid.set_position_xy(position);
+        }
+
+        fn refit_entries_to_shape(&self, shape: &ensogl::display::scene::Shape) {
+            // === Resize ===
+
+            let width = shape.width / Columns::LEN as f32;
+            let size = Vector2(width, ENTRY_HEIGHT);
+            self.grid.set_entries_size(size);
         }
     }
 
-    // === SampleComponent ===
 
-    #[derive(Clone, CloneRef, Debug)]
-    #[allow(missing_docs)]
+    // === Getter `impl`s ===
+
+    impl Model {
+        /// Returns the number of [`Grid`] rows needed to display all the data in this [`Model`].
+        ///
+        /// The number of rows is equal to the number of [`Project`]s in the [`Model`] plus one,
+        /// because the first row is the header row.
+        ///
+        /// [`Grid`]: ::crate::projects_grid::Grid
+        /// [`Model`]: ::crate::projects_grid::Model
+        /// [`Project`]: ::enso_cloud_view::project::Project
+        fn rows(&self) -> usize {
+            self.projects.borrow().len() + 1
+        }
+    }
+
+
+
+    // ============
+    // === View ===
+    // ============
+
+    #[derive(Clone, Debug)]
     pub struct View {
-        pub frp: Frp,
-        model:   Model,
+        pub frp:   Frp,
+        pub model: Model,
     }
 
+
+    // === Main `impl` ===
+
     impl View {
-        pub fn new(app: &Application) -> Self {
+        fn new(app: &Application) -> Self {
             let frp = Frp::new();
             let network = &frp.network;
             let model = Model::new(app);
 
-            let input = &frp.private.input;
-            let output = &frp.private.output;
+            let grid = &model.grid;
+            let input = &frp.public.input;
+            let out = &frp.public.output;
+            let scene = &app.display.default_scene;
 
             frp::extend! { network
 
-                eval input.set_size((size) model.set_size(*size));
-                output.size <+ input.set_size;
 
-                eval_ model.base_shape.events.mouse_over({
-                    model.set_size(Vector2::new(300.0,300.0));
-                    tracing::info!("Mouse over!")
-                });
-                eval_ model.base_shape.events.mouse_out({
-                    model.set_size(Vector2::new(200.0,200.0));
-                    tracing::info!("Mouse out!")
-                });
-                eval_ model.base_shape.events.mouse_down([] tracing::info!("Mouse down!"));
-                eval_ model.base_shape.events.mouse_up([] tracing::info!("Mouse up!"));
+                // === Model I/O ===
+
+                // FIXME [NP]: How do we get rid of this clone?
+                eval input.set_projects((projects) model.set_projects((*projects).clone()));
```

Comment:
take slice there

Reply:
Can't since we need to swap out the contents of a refcell but since this is now a SharedVec this clone is cheap